include/k4arecord/playback.h

```diff
@@ -370,6 +601,190 @@ K4ARECORD_EXPORT k4a_stream_result_t k4a_playback_get_next_imu_sample(k4a_playba
+ * Handle obtained by k4a_playback_open().
+ *
+ * \param track_name
+ * The name of the track to read the previous data block from.
+ *
+ * \param data_block_handle
+ * The location to write the data block.
+ *
+ * \returns
+ * ::K4A_STREAM_RESULT_SUCCEEDED if a sample is returned, or ::K4A_STREAM_RESULT_EOF if the start of the recording is
+ * reached. All other failures will return ::K4A_STREAM_RESULT_FAILED.
+ *
+ * \relates k4a_playback_t
+ *
+ * \remarks
+ * k4a_playback_get_previous_data_block() always returns the data block before the most recently returned data block for
+ * a particular track.
+ *
+ * \remarks
+ * If a call was made to to k4a_playback_get_next_data_block() which returned ::K4A_STREAM_RESULT_EOF, then the playback
+ * position is at the end of the recording and calling k4a_playback_get_previous_data_block() with the same track will
+ * return the last data block in the track.
+ *
+ * \remarks
+ * The first call to k4a_playback_get_previous_data_block() after k4a_playback_seek_timestamp() will return the
+ * data block closest to the seek time with a timestamp less than the seek time.
+ *
+ * \remarks
+ * If the call is successful, callers must call k4a_playback_data_block_release() to return the allocated memory for
+ * data_block_handle.
+ *
+ * \remarks
+ * k4a_playback_get_previous_data_block() cannot be used with the built-in tracks: "COLOR", "DEPTH", etc...
+ * k4a_playback_track_is_builtin() can be used to determine if a track is a built-in track.
+ *
+ * \xmlonly
+ * <requirements>
+ *   <requirement name="Header">playback.h (include k4arecord/playback.h)</requirement>
+ *   <requirement name="Library">k4arecord.lib</requirement>
+ *   <requirement name="DLL">k4arecord.dll</requirement>
+ * </requirements>
+ * \endxmlonly
+ */
+K4ARECORD_EXPORT k4a_stream_result_t k4a_playback_get_previous_data_block(k4a_playback_t playback_handle,
+                                                                          const char *track_name,
+                                                                          k4a_playback_data_block_t *data_block_handle);
+
+/** Get the timestamp of a data block in microseconds.
+ *
+ * \param data_block_handle
+ * Handle obtained by k4a_playback_get_next_data_block() or k4a_playback_get_previous_data_block().
+ *
+ * \returns
+ * Returns the timestamp of the data block. If the \p data_block_handle is invalid this function will return 0.
+ * It is also possible for 0 to be a valid timestamp originating from the beginning of a recording.
+ *
+ * \relates k4a_playback_data_block_t
+ *
+ * \xmlonly
+ * <requirements>
+ *   <requirement name="Header">playback.h (include k4arecord/playback.h)</requirement>
+ *   <requirement name="Library">k4arecord.lib</requirement>
+ *   <requirement name="DLL">k4arecord.dll</requirement>
+ * </requirements>
+ * \endxmlonly
+ */
+K4ARECORD_EXPORT uint64_t k4a_playback_data_block_get_timestamp_usec(k4a_playback_data_block_t data_block_handle);
```

Comment:
We now have device and and system timestamps. This should be namespaced with the timestamp being returned.